Dozer/DozerIcons.swift

```diff
@@ -86,6 +86,12 @@ public final class DozerIcons {
         }
     }
 
+    public var enableIconAndMenu: Bool = Defaults[.showIconAndMenuEnabled] {
+        didSet {
+            Defaults[.showIconAndMenuEnabled] = self.enableIconAndMenu
```

Comment:
When unchecking "Show Dozer menu and dock icon whilst interacting" then dock icon does not disappear. Only when restarting Dozer. The following would fix that. 
```suggestion
            Defaults[.showIconAndMenuEnabled] = self.enableIconAndMenu
            if self.enableIconAndMenu == false {
                _ = DozerIcons.toggleDockIcon(showIcon: false)
                appDelegate.preferencesWindowController.show(preferencePane: .general)
            }
```
What do you think about showing the preferences window again? It gliches shortly, but it may be better than it just disappearing.

Reply:
Glad you like it! It's been a pleasure to do this PR. 

I'll implement the fix. 

I'm not sure what the preference window issue is?

Reply:
Nevermind about the issue about the preference window. No problem there.